src/class-bulk-task.php

```diff
@@ -157,6 +166,130 @@ public function filter__posts_where( $where, $query ) {
 		return $where;
 	}
 
+	/**
+	 * Manipulate the WHERE clause of a bulk task term query to batch by
+	 * term_taxonomy_id. We're using term_taxonomy_id rather than term_id because
+	 * they're less likely to span very large ranges.
+	 *
+	 * This checks the object hash to ensure that we don't manipulate any other
+	 * queries that might run during a bulk task.
+	 *
+	 * @link https://github.com/WordPress/wordpress-develop/blob/6.1/src/wp-includes/class-wp-term-query.php#L731
+	 *
+	 * @param array $clauses Associative array of the clauses for the query.
+	 * @return string Filtered array with our batching added to the WHERE clause.
+	 */
+	public function filter__terms_where( $clauses ) {
+		if ( ! empty( $this->query ) && spl_object_hash( $this->query ) === $this->object_hash ) {
+			$clauses['where'] .= sprintf(
+				' AND tt.term_taxonomy_id > %d AND tt.term_taxonomy_id <= %d',
+				$this->min_id,
+				$this->min_id + $this->stepping
+			);
+		}
+
+		return $clauses;
+	}
+
+	/**
+	 * Loop through any number of objects efficiently with a callback, and output
+	 * the progress.
+	 *
+	 * @param array    $args Array of args to pass to query.
+	 * @param callable $callable Callback function to invoke for each object.
+	 *                           The callable will be passed an object of the
+	 *                           specified type.
+	 * @param string   $object_type Type of object to query.
+	 */
+	public function run( array $args, callable $callable, string $object_type = 'wp_post' ): void {
+		if ( ! method_exists( $this, "run_${object_type}_query" ) ) {
+			return;
+		}
+
+		call_user_func( [ $this, "run_${object_type}_query" ], $args, $callable );
+	}
+
+	/**
+	 * Loop through any number of terms efficiently with a callback, and output
+	 * the progress.
+	 *
+	 * @param array    $args {
+	 *     WP_Term_Query args. Some have overridden defaults, and some are fixed.
+	 *     Anything not mentioned below will operate as normal.
+	 *
+	 *     @type string $order                  Always 'ASC'.
+	 *     @type string $orderby                Always 'term_id'.
+	 *     @type int    $number                 Defaults to 0 (all).
+	 *     @type bool   $update_term_meta_cache Always false.
+	 * }
+	 * @param callable $callable Callback function to invoke for each post.
+	 *                           The callable will be passed a post object.
+	 */
+	public function run_wp_term_query( array $args, callable $callable ): void {
+		global $wpdb;
+
+		// Apply default arguments.
+		$args = wp_parse_args(
+			$args,
+			[
+				'number' => 0,
+			],
+		);
+
+		// Force some arguments and don't let them get overridden.
+		$args['order']                  = 'ASC';
+		$args['orderby']                = 'term_id';
+		$args['update_term_meta_cache'] = false;
```

Comment:
Suggestion: Those could be moved to `wp_parse_args` and set as default values there.

Reply:
Disregard this comment. I see why it doesn't use that function now.